modules/highgui/CMakeLists.txt

```diff
@@ -275,6 +274,11 @@ if(MSVC AND NOT BUILD_SHARED_LIBS AND BUILD_WITH_STATIC_CRT)
   set_target_properties(${the_module} PROPERTIES LINK_FLAGS "/NODEFAULTLIB:atlthunk.lib /NODEFAULTLIB:atlsd.lib /NODEFAULTLIB:libcmt.lib /DEBUG")
 endif()
 
+if(HAVE_WIN32UI AND HAVE_OPENGL AND OPENGL_LIBRARIES)
```

Comment:
> HAVE_WIN32UI

I would prefer to use `OPENCV_HIGHGUI_BUILTIN_BACKEND` instead as we have built-in and plugin backends in OpenCV 4.x+. See #21303